handlers.go

```diff
@@ -54,23 +54,30 @@ func receiveHandler(producer *kafka.Producer, serializer Serializer) func(c *gin
 			return
 		}
 
-		metrics, err := processWriteRequest(&req)
+		metricsPerTopic, err := processWriteRequest(&req)
 		if err != nil {
 			c.AbortWithStatus(http.StatusInternalServerError)
 			logrus.WithError(err).Error("couldn't process write request")
 			return
 		}
 
-		for _, metric := range metrics {
-			err := producer.Produce(&kafka.Message{
-				TopicPartition: kafkaPartition,
-				Value:          metric,
-			}, nil)
+		for topic, metrics := range metricsPerTopic {
+			t := &topic
```

Comment:
This is not correct. Here is an example: https://play.golang.org/p/lgiyXGh2fB8

It should be:

```go
		for topic, metrics := range metricsPerTopic {

			t := topic // getting the value, not the reference

			part := kafka.TopicPartition{
				Partition: kafka.PartitionAny,
				Topic:     &t, // setting the reference of the local variable
			}
```

With this last change, the PR looks good to me. Thanks for your contribution @jbguerraz 🎉 !

Reply:
thank you!